### PR TITLE
parsing(c_parser): Added parsing code to handle the assignment case where character is assigned to integer

### PR DESCRIPTION
**Pull request by smitgajjar**
#### Brief description of what is fixed or changed
Added the case where character is assigned to an integer
- Added code in the function transform_character_literal() to return appropriate string
- Handled the AST clang nodes in transform_var_decl()
-  Added tests for the same.

#### Other comments
Consider the following case :
<pre>
In [1]: from sympy.parsing.sym_expr import SymPyExpression                      

In [2]: src = """ 
   ...: int a='1', b='d'; 
   ...: """                                                                     

In [3]: a=SymPyExpression(src, 'c')                                             

In [4]: a.return_expr()                                                         
Out[4]: 
[Declaration(Variable(a, type=integer, value=49)), Declaration(Variable(b, typ
e=integer, value=100))]

</pre>

Before: Raised NotImplementedError
Now: Assigns correct ascii value to the integer value as expected in C and successfully gets parsed to sympy expression as shown above.

#### Release Notes

- parsing
    - Added the assignment case where character is assigned to an integer in C parser

### CODE REVIEW

File: sympy/parsing/c/c_parser.py [hunk 1]
```diff
@@ -478,16 +488,29 @@ def transform_string_literal(self, node):
...
-            #    value = next(node.get_tokens()).spelling
-            #except (StopIteration, ValueError):
+            """Transformation function for character literal
+
+            Used to get the value of the given character literal.
+
+            Returns
+            =======
+
+            val : value
```
Comment (Smit-create) [inline sympy/parsing/c/c_parser.py]: I think this should be
```
val : str
     Its description
```

Comment (smitgajjar) [reply sympy/parsing/c/c_parser.py]: Thanks for suggestion! I will change it as requested.

File: sympy/parsing/c/c_parser.py [hunk 2]
```diff
@@ -478,16 +488,29 @@ def transform_string_literal(self, node):
...
+            try:
+               value = next(node.get_tokens()).spelling
+            except (StopIteration, ValueError):
                 # No tokens
-            #    value = node.literal
-            #val = [type, value]
-            #return val
-            pass
+               value = node.literal
+            return str(value[1])
```
Comment (Sc0rpi0n101) [inline sympy/parsing/c/c_parser.py]: why does this have to be `value[1]`?
Wouldn't `value` have the character literal that can directly be returned?

File: sympy/parsing/c/c_parser.py [hunk 3]
```diff
@@ -256,6 +256,16 @@ def transform_var_decl(self, node):
                     ).as_Declaration(
                         value = val
                     )
+                elif (child.kind == cin.CursorKind.CHARACTER_LITERAL
```
Comment (Sc0rpi0n101) [inline sympy/parsing/c/c_parser.py]: Can you add a comment here describing when it will be called?

**Review by Sc0rpi0n101 — COMMENTED** [review]
I've left a few comments. 
Otherwise, it looks good.

File: sympy/parsing/c/c_parser.py (hunk 2 above)
Comment (smitgajjar) [reply sympy/parsing/c/c_parser.py]: Actually, `node.literal` returns a str having value `'<char>'`, whereas we only want <char>. So, it is a sort of workaround.

File: sympy/parsing/c/c_parser.py (hunk 3 above)
Comment (smitgajjar) [reply sympy/parsing/c/c_parser.py]: Sure, will add it in the next commit.

File: sympy/parsing/c/c_parser.py (hunk 2 above)
Comment (Sc0rpi0n101) [reply sympy/parsing/c/c_parser.py]: ok

**Comment by Sc0rpi0n101** [comment]
I think the rest looks good.

Can you also squash your commits into one with a description of all the changes?

**Comment by smitgajjar** [comment]
I guess, this longer commit message looks weird! Anyways, did the squash! :)

**Comment by smichr** [comment]
> commit message looks weird

If you add a blank line between title of commit and commit message it will look better:
```
title blah blah

comments
```

**Comment by smitgajjar** [comment]
Alright, thanks for the help, I will do it now!

**Pull request merged**